docs/root/version_history/current.rst

```diff
@@ -49,6 +49,7 @@ New Features
 * http: added support for :ref:`proxy_status_config <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.proxy_status_config>` for configuring `Proxy-Status <https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-proxy-status-08>`_ HTTP response header fields.
 * http: make consistent custom header format fields ``%(DOWN|DIRECT_DOWN|UP)STREAM_(LOCAL|REMOTE)_*%`` to provide all combinations of local & remote addresses for upstream & downstream connections.
 * http3: downstream HTTP/3 support is now GA! Upstream HTTP/3 also GA for specific deployments. See :ref:`here <arch_overview_http3>` for details.
+* http3: supports upstream HTTP/3 retries. Automatically retry 0-RTT safe requests if they are rejected because they are sent too early. And automatically retry 0-RTT safe requests if connect attempt fails later on and the cluster is configured with TCP fallback. And add retry on ``alt-protocols-post-connect-failure`` policy which allows retry of failed HTTP/3 requests with TCP fallback even after handshake if the cluster is configured with TCP fallback.
```

Comment:
Is there somewhere you can link to (RFC, etc.) that explains "0-RTT safe requests?"

Also, I think alt-protocols-post-connect-failure is out of date and needs updating? See my other comment also about additional docs on this wherever the other retry docs are.

Reply:
There's a link to down in http/utility.cc to https://www.rfc-editor.org/rfc/rfc7231#section-4.2.1 which defines safe methods. I guess we could also link to https://datatracker.ietf.org/doc/html/rfc8470 which talks about using safe methods for early data. We could mention this here, perhaps?